packages/webcrypto-ed25519-polyfill/src/secrets.ts

```diff
@@ -131,10 +131,53 @@ export function verifyPolyfill(key: CryptoKey, signature: BufferSource, data: Bu
     if (key.type !== 'public' || !key.usages.includes('verify')) {
         throw new DOMException('Unable to use this key to verify', 'InvalidAccessError');
     }
-    const publicKeyBytes = ed25519.getPublicKey(getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key));
+    const storedBytes = getSecretKeyBytes_INTERNAL_ONLY_DO_NOT_EXPORT(key);
     try {
-        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), publicKeyBytes);
+        return ed25519.verify(bufferSourceToUint8Array(signature), bufferSourceToUint8Array(data), storedBytes);
     } catch {
         return false;
     }
 }
+
+export function importKeyPolyfill(
+    format: KeyFormat,
+    keyData: BufferSource,
+    extractable: boolean,
+    keyUsages: readonly KeyUsage[]
+): CryptoKey {
+    if (keyUsages.length === 0) {
+        throw new DOMException('Usages cannot be empty when creating a key.', 'SyntaxError');
+    }
+    if (keyUsages.some(usage => PROHIBITED_KEY_USAGES.has(usage))) {
+        throw new DOMException('Unsupported key usage for an Ed25519 key.', 'SyntaxError');
+    }
+
+    const bytes = bufferSourceToUint8Array(keyData);
+
+    switch (format) {
+        case 'raw': {
+            const publicKey = {
+                [Symbol.toStringTag]: 'CryptoKey',
+                algorithm: Object.freeze({ name: 'Ed25519' }),
+                extractable: true,
+                type: 'public' as KeyType,
+                usages: Object.freeze(keyUsages.filter(usage => usage === 'verify')) as KeyUsage[],
+            };
+
+            const cache = (storageKeyBySecretKey_INTERNAL_ONLY_DO_NOT_EXPORT ||= new WeakMap());
+            cache.set(publicKey, bytes);
+
+            return publicKey;
+        }
+        case 'pkcs8': {
```

Comment:
You have to also validate that the usages includes _only_ `'sign'`. You can't have PKCS#8 public keys. Trying to import a pkcs-encoded key with the `'verify'` usage should throw `Unsupported key usage for a Ed25519 key`.